src/iexamine.cpp

```diff
@@ -3465,3 +3389,47 @@ iexamine_function iexamine_function_from_string(std::string const &function_name
     debugmsg("Could not find an iexamine function matching '%s'!", function_name.c_str());
     return &iexamine::none;
 }
+
+hack_result iexamine::hack_attempt(player *p) {
+    if (p->has_trait("ILLITERATE")) {
+        return HACK_UNABLE;
+    }
+    bool using_electrohack = (p->has_amount("electrohack", 1) &&
+                              query_yn(_("Use electrohack?")));
+    bool using_fingerhack = (!using_electrohack && p->has_bionic("bio_fingerhack") &&
+                             p->power_level > 0 &&
+                             query_yn(_("Use fingerhack?")));
+
+    if ( using_electrohack || using_fingerhack ) {
```

Comment:
Might be better to invert  this condition and return early with `HACK_UNABLE`. Currently the main part of this function is needlessly intended inside the if-branch. It would also be more consistent with the other precondition check (illiterate) above.